packages/frontend/components/primitives/string-cutter/component.tsx

```diff
@@ -0,0 +1,52 @@
+import type { FC } from 'react';
+import { useEffect, useState, useRef } from 'react';
+import type { StringCutterProps } from './types';
+
+import * as styles from './styles';
+
+const StringCutter: FC<StringCutterProps> = ({ text }) => {
+  const [isCutted, setIsCutted] = useState(false);
+  const [isHover, setIsHover] = useState(false);
+  const cutterRef = useRef(null);
+
+  useEffect(() => {
+    const cutterElem = cutterRef.current;
+    if (cutterElem) {
+      setIsCutted(cutterElem.offsetWidth < cutterElem.scrollWidth);
+    }
+  }, [cutterRef]);
```

Comment:
It will not work if we resize the screen. cutterRef immutable itself. I suggest you not to store isHover. You can check if elelement is cut on hover and setState. Then, render tooltip if only isCut.